File: Current/Product/Production/Nant/VSS.Tasks/VSSAddProject.vb
Option Explicit On 
Option Strict On

Imports NAnt.Core
Imports NAnt.Core.Attributes
Imports NAnt.Contrib.Tasks.SourceSafe

<TaskName("vssaddproject")> _
Public Class VSSAddProject
    Inherits BaseTask

    Private _ProjectName As String
    Private _Comment As String

    <TaskAttributeAttribute("comment", Required:=False)> _
    Public Property Comment() As String
        Get
            Return _Comment
        End Get
        Set(ByVal Value As String)
            _Comment = Value
        End Set
    End Property

    <TaskAttributeAttribute("project", Required:=True)> _
    Public Property ProjectName() As String
        Get
            Return _ProjectName
        End Get
        Set(ByVal Value As String)
            _ProjectName = Value
        End Set
    End Property

    Protected Overrides Sub ExecuteTask()
        MyBase.Open()
        MyBase.Item.NewSubproject(Me.ProjectName, Me.Comment)
    End Sub

    Public Sub test()
        Try
            Me.Comment = "Testing"
            Me.DBPath = New IO.FileInfo("C:\Source Safe DataBases\Test\srcsafe.ini")
            Me.Password = "password"
            Me.UserName = "build"
            Me.Path = "$/"
            Me.ProjectName = "Test"
            Me.ExecuteTask()
        Catch ex As Exception
            Stop
        End Try

    End Sub

End Class

File: tags/0.8.0.82/Product/Production/Nant/NAnt.Contrib.Tasks/Tasks/SourceSafe/VSSAddProject.cs
using System;

using NAnt.Core;
using NAnt.Core.Attributes;
using NAnt.Contrib.Tasks.SourceSafe;

namespace NAnt.Contrib.Tasks.SourceSafe
{
    [TaskName("vssaddproject")]
    public class VSSAddProject : BaseTask
    {
        #region Fields

        private string _Comment;

        private string _ProjectName;

        #endregion

        #region Properties

        [TaskAttributeAttribute("comment", Required = false)]
        public string Comment
        {
            get { return _Comment; }
            set { _Comment = value; }
        }

        [TaskAttributeAttribute("project", Required = true)]
        public string ProjectName
        {
            get { return _ProjectName; }
            set { _ProjectName = value; }
        }

        #endregion

        #region Public Methods

        public void test()
        {
            try
            {
                this.Comment = "Testing";
                this.DBPath = new System.IO.FileInfo("C:\\Source Safe DataBases\\Test\\srcsafe.ini");
                this.Password = "password";
                this.UserName = "build";
                this.Path = "$/";
                this.ProjectName = "Test";
                this.ExecuteTask();
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debugger.Break();
            }
        }

        #endregion

        #region Protected Methods

        protected override void ExecuteTask()
        {
            base.Open();
            base.Item.NewSubproject(this.ProjectName, this.Comment);
        }

        #endregion

    }
}